doc/pmon/pmon dependency.md

```diff
@@ -0,0 +1,43 @@
+# Root causes:
+## when dockers stop
+For any shutdown flow, which means all dockers are stopped in order, pmon docker often stops after syncd docker has stopped, causing pmon docker fail to release sx_core resources and leaving sx_core in a bad state. The related logs are like the following:
+
+    INFO syncd.sh[23597]: modprobe: FATAL: Module sx_core is in use.
+    INFO syncd.sh[23597]: Unloading sx_core[FAILED]
+    INFO syncd.sh[23597]: rmmod: ERROR: Module sx_core is in use
+
+## when dockers restart without reboot
+In the flows like "config reload" and "service swss restart", the failure cause further consequences:
+1. sx_core initialization error with error message like "sx_core: create EMAD sdq 0 failed. err: -16"
+2. syncd fails to execute the create switch api with error message "syncd_main: Runtime error: :- processEvent: failed to execute api: create, key: SAI_OBJECT_TYPE_SWITCH:oid:0x21000000000000, status: SAI_STATUS_FAILURE"
+3. swss fails to call SAI API "SAI_SWITCH_ATTR_INIT_SWITCH", which causes orchagent to restart.
+This will introduce an extra 1 or 2 minutes for the system to be available, failing related test cases. In this sense, the issue should be solved for the flow.
+
+## reboot flow
+In the reboot flows including "reboot", "fast-reboot" and "warm-reboot" this failure doesn't have further negative effects since the system has already rebooted. In addition, "warm-reboot" requires the system to be shutdown as soon as possible to meet the GR time restriction of both BGP and LACP. "fast-reboot" also requires to meet the GR time restriction of BGP which is longer than LACP. In this sense, any unnecessary steps should be avoided. It's better to keep those flows untouched.
```

Comment:
although it have no real negative impact in the reboot case,  one concern is that the error log could trigger log analyzer failure and cause the whole test to fail.

Reply:
This can be addressed by using "ignore" in log analyzer.